specification/purview/resource-manager/Microsoft.Purview/preview/2024-04-01-preview/purview.json

```diff
@@ -1764,7 +1830,20 @@
           ],
           "type": "string",
           "x-ms-enum": {
-            "name": "publicNetworkAccess",
+            "name": "PublicNetworkAccess",
+            "modelAsString": true
+          }
+        },
+        "tenantEndpointState": {
+          "description": "Gets or sets the state of tenant endpoint.",
+          "enum": [
+            "NotSpecified",
```

Comment:
I am curious about whether you intend to allow setting it to the NotSpecified state, and whether that is what it defaults to - you might want to update the description to clarify

Reply:
Now that I see it, I also feel a concern that you allow setting publicNetworkAccess to NotSpecified and that I'm not familiar with any other RPs doing that for private links Is that a 'new' invention from your RP?

Or am I just out of date with my knowledge of what network team recommended for this?

Reply:
“Not specified” is for our internal logic to differentiate between existing accounts vs newly created accounts. But once the call is received, it’ll be set to enabled/disabled based on the internal checks.

Reply:
So, do I understand correctly that you would never actually want 'NotSpecified' to be sent in the request, and never want to return it in the response?
If I got that right, it seems quite confusing to add it to the enumeration!

Reply:
- If the value is explicitly given as 'Disabled'/'Enabled' - we honor it.
- If it is 'NotSpecified' - 
  - If it is a new account creation after the feature launch- it will be enabled by default.
  - If it is updating an exisiting account - it'll be disabled by default till the customer performs mandatory setup before enabling it.
  - If a value is continued to be 'NotSpecified'- it helps us differentiate that its an existing account that did not get updated after the feature launch.

Reply:
@ddumesnil-microsoft
But why do you need a 'NotSpecified' value at all, in json format? Why not just have the field be modeled as optional? 
Since you seem to say it won't return NotSpecified in the response. It will return Disabled or Enabled, is that right?

Reply:
NotSpecified could be returned in the response. It is the value that is used for accounts that existed before the feature launched (as new accounts are Enabled by default and could otherwise be updated to Disabled) and have not undergone any updates (as that would set it to Disabled by default).

This allows us to have easier tracked values as the value will be present for all accounts rather than using `null` as the NotSpecified value.